airflow/models/connection.py

```diff
@@ -408,3 +408,15 @@ def get_connection_from_secrets(cls, conn_id: str) -> 'Connection':
                 )
 
         raise AirflowNotFoundException(f"The conn_id `{conn_id}` isn't defined")
+
+    def __eq__(self, other: 'Connection'):
```

Comment:
It's not recommended to annotate `__eq__` with a concrete type. The recommended approach is something like this:

```python
def __eq__(self, other: object) -> bool:  # I think Any would work as well.
    if not isinstance(other, Connection):
        return NotImplemented
    # Actual comparison logic...
```

See also:

https://docs.python.org/3/library/numbers.html#implementing-the-arithmetic-operations
https://stackoverflow.com/questions/37557411/why-does-defining-the-argument-types-for-eq-throw-a-mypy-type-error